examples/platforms/nrf528xx/src/radio.c

```diff
@@ -451,7 +461,10 @@ otError otPlatRadioReceive(otInstance *aInstance, uint8_t aChannel)
         nrf5FemEnable();
     }
 
-    nrf_802154_tx_power_set(sDefaultTxPower);
+    transmitPower = sDefaultTxPower < sMaxTxPowerTable[channel - OT_RADIO_2P4GHZ_OQPSK_CHANNEL_MIN]
+                        ? sDefaultTxPower
+                        : sMaxTxPowerTable[channel - OT_RADIO_2P4GHZ_OQPSK_CHANNEL_MIN];
```

Comment:
1. Sanity check for `channel` before use or use `nrf5GetChannelMaxTransmitPower` instead?
2. Check `sPowerInitialized`?
```suggestion
    int8_t channelMaxTxPower = nrf5GetChannelMaxTransmitPower(channel);
    
    transmitPower = sPowerInitialized && sDefaultTxPower < channelMaxTxPower
                        ? sDefaultTxPower
                        : channelMaxTxPower;
```